src/error.rs

```diff
@@ -10,4 +11,13 @@ pub enum ReconcileError {
 
     #[error("There are no ports set on this LoadBalancer")]
     NoPortsSet,
+
+    #[error("The provided cloud provisioner was not found in the cluster")]
+    CloudProvisionerNotFound,
+    #[error("The secret keys for the cloud provisioner were not found in the cluster")]
+    CloudProvisionerSecretNotFound,
+
+    #[error("The operator has encountered an error: {0}")]
+    OperatorError(#[from] color_eyre::Report),
```

Comment:
This is bad naming, this entire enum is full of OperatorErrors lmfao

maybe `UnknownError` or `OtherError`?